src/main/java/com/echobox/cache/PeriodicCacheWithCallback.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.echobox.cache;
+
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.lang3.NotImplementedException;
+
+import java.io.Serializable;
+import java.util.function.Supplier;
+
+/**
+ * Periodically executes the callback to get the latest value.
+ * If it's within the specified period then get the cached value.
+ * If the callback fails still get the cached value and update the cache next time
+ * @param <T> return type of cache object
+ * @author Daniyar
+ */
+public class PeriodicCacheWithCallback<T extends Serializable> {
```

Comment:
```suggestion
public class CacheWithSupplierFailover<T extends Serializable> {
```

Thinking about the use case this proposed name feels clearer, as it's the failover part we're specifically covering.